python/paddle/fluid/dygraph/dygraph_to_static/ast_utils.py

```diff
@@ -31,12 +31,60 @@
 FALSE_FUNC_PRFIX = 'false_fn'
 
 
+class IsControlFlowIfVisitor(gast.NodeTransformer):
+    """
+    Judge whether the node.test from Dygraph code dependent on paddle Tensor.
+    If does, it should satisfy:
+        1. must involve at least one var whose type is Tensor.
+        2. the Tensor var should call `.numpy()[]` interface or Tensor.shape is [1].
+        3. involve Tensor.shape[i] and the shape[i] is unknown in compile time.
+    The following examples should not be considered as control_flow_if:
+        1. `if Tensor_var` or `if Tensor_var is None`
+        2. if Tensor.shape[i] is determined with fixed value (not -1 or None)
+
+    Note: pred in ConditionalBlock require variable, which means all vars should be Tensor
+          or transformed into Tensor, like fill_constant(shape=[1], dtype='int32', value=Tensor.shape[i]).
+
+    TODO: 1. need to deal with `tensor.shape[i]` which need eval the data of shape[i],
+             because reshape_op maybe called before this statement.
```

Comment:
typo:
need eval -> need to eval
 maybe -> may be.

Reply:
Fixed, thanks.